src/FondOfSpryker/Glue/ProductListsRestApi/Processor/ProductLists/ProductListsReader.php

```diff
@@ -0,0 +1,260 @@
+            ->setUuid($restRequest->getResource()->getId());
+
+        $productListResponseTransfer = $this->productListsClient
+            ->findProductListByUuid($productListTransfer);
+
+        if (!$productListResponseTransfer->getIsSuccessful()) {
+            return $this->restApiError->addProductListNotFoundError($restResponse);
+        }
+
+        if (!$this->productListIsFromCustomerOrCompanyUser($productListResponseTransfer->getProductList(), $restRequest->getRestUser())) {
+            return $this->restApiError->addProductListNoPermissionError($restResponse);
+        }
+
+        $restProductListsAttributesTransfer = $this->productListsMapper
+            ->mapProductListTransferToRestProductListResponseAttributesTransfer(
+                $productListResponseTransfer->getProductList(),
+                new RestProductListsAttributesTransfer()
+            );
+
+        return $this->addRestProductListsAttributesTransferToResponse($restProductListsAttributesTransfer, $restResponse);
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\ProductListTransfer $productListTransfer
+     * @param \Generated\Shared\Transfer\RestUserTransfer|null $restUserTransfer
+     *
+     * @return bool
+     */
+    protected function productListIsFromCustomerOrCompanyUser(
+        ProductListTransfer $productListTransfer,
+        ?RestUserTransfer $restUserTransfer
+    ): bool {
+        if ($restUserTransfer === null) {
+            return false;
+        }
+
+        if ($this->productListCustomerRelationHasRestUser(
+            $productListTransfer->getProductListCustomerRelation(),
+            $restUserTransfer
+        )) {
+            return true;
+        }
+
+        if ($this->productListCompanyUserRelationshipHasRestUserCompanyId(
+            $productListTransfer->getProductListCompanyRelation(),
+            $restUserTransfer
+        )) {
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\ProductListCustomerRelationTransfer|null $productListCustomerRelationTransfer
+     * @param \Generated\Shared\Transfer\RestUserTransfer $restUserTransfer
+     *
+     * @return bool
+     */
+    protected function productListCustomerRelationHasRestUser(
+        ?ProductListCustomerRelationTransfer $productListCustomerRelationTransfer,
+        RestUserTransfer $restUserTransfer
+    ): bool {
+        if ($productListCustomerRelationTransfer === null) {
+            return false;
+        }
+
+        return in_array($restUserTransfer->getSurrogateIdentifier(), $productListCustomerRelationTransfer->getCustomerIds(), true);
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\ProductListCompanyRelationTransfer|null $productListCompanyRelationTransfer
+     * @param \Generated\Shared\Transfer\RestUserTransfer $restUserTransfer
+     *
+     * @return bool
+     */
+    protected function productListCompanyUserRelationshipHasRestUserCompanyId(
```

Comment:
"isProductListAssignedToCompanyUser" is better than "productListCompanyUserRelationshipHasRestUserCompanyId"